src/libraries/System.Text.RegularExpressions/tests/RegexExperiment.cs

```diff
@@ -0,0 +1,193 @@
+            genUnicode.Invoke(null, new object[] { tmpWorkingDir });
+        }
+
+        /// <summary>
+        /// View the regex as a DFA in DGML format in VS.
+        /// </summary>
+        /// <param name="r"></param>
+        private static void ViewDGML(Regex regex, int bound = -1, bool hideStateInfo = true, bool addDotStar = false, bool inReverse = false, bool onlyDFAinfo = false, string name = "DFA", int maxLabelLength = 20)
+        {
+            StringWriter sw = new StringWriter();
+            SaveDGML(regex, sw, bound, hideStateInfo, addDotStar, inReverse, onlyDFAinfo, maxLabelLength);
+            File.WriteAllText(string.Format("{1}{0}.dgml", inReverse ? name + "r" : (addDotStar ? name + "1" : name), dgmloutdirectory), sw.ToString());
+        }
+
+        /// <summary>
+        /// The intent is that this method is run in realease build for lightweight performance testing.
+        /// One can e.g. open the outputfile in emacs with AUTO-REVERT-ON in order to follow the progress in real time.
+        /// It will print timing info and match info for both DFA and Compiled option.
+        /// Place sample regexes in the regexesfile (one per line) and sample input in inputfile.
+        /// </summary>
+        //[Fact]
+        private void TestRunPerformance()
+        {
+            string input = File.ReadAllText(inputfile);
+            string[] rawregexes = File.ReadAllLines(regexesfile);
+            WriteOutput("\n========= date:{0} =========\n", System.DateTime.Now);
+            int totDFA = 0;
+            int totCOM = 0;
+            int tDFA = 0;
+            int tCOM = 0;
+            for (int i = 0; i < rawregexes.Length; i++)
+            {
+                var rawregex = rawregexes[i];
+                Regex re = new(rawregex, DFA, new TimeSpan(0, 0, 10));
+                Regex reC = new(rawregex, RegexOptions.Compiled, new TimeSpan(0, 0, 10));
+                WriteOutput("\n--- Regex:{0}\n", i);
+                //-------------------
+                //-- measure DFA
+                //-------------------
+                tDFA = MeasureMatchTime(re, "DFA", input);
+                //-------------------
+                //-- measure COMPILED
+                //-------------------
+                tCOM = MeasureMatchTime(reC, "COM", input);
+                //ignore the cases when one times out
+                if (tDFA >= 0 && tCOM >= 0)
+                {
+                    totDFA += tDFA;
+                    totCOM += tCOM;
+                }
+            }
+            WriteOutput("\ntotal time: DFA:{0}ms, COM:{1}ms\n", totDFA, totCOM);
+        }
+
+        private static int MeasureMatchTime(Regex re, string tag, string input)
+        {
+            File.AppendAllText(outputfile, tag + ": warmup:...");
+            Stopwatch sw = new Stopwatch();
+            int t = 0;
+            try
+            {
+                sw.Start();
```

Comment:
nit, you can write `var sw = Stopwatch.StartNew();`

Reply:
ok, will do